src/ota.c

```diff
@@ -88,10 +96,49 @@ const struct golioth_ota_component *golioth_ota_find_component(
     return found;
 }
 
+static void ota_manifest_timer_expiry(golioth_sys_timer_t timer, void *user_arg)
+{
+    struct manifest_get_args *get_args = user_arg;
+
+    enum golioth_status status = golioth_ota_get_manifest_async(get_args->client,
+                                                                get_args->cb,
+                                                                get_args->arg);
+
+    if (GOLIOTH_OK != status)
+    {
+        GLTH_LOGE(TAG, "Periodic OTA manifest fetch failed");
+    }
```

Comment:
Because observation establishment / cancellation is handled internally to the client, we can't easily cancel this timer, so I expect that if the client is stopped then we'll see this error logged whenever the timer fires? I think that is okay for the time-being, but if we move more towards the harmonic polling you've described in the future I suspect we could move to a single internally managed timer and avoid issuing requests when the client is stopped.